### DISCUSSION TITLE
Select component typeahead doesn't work when dropdown open?

### DISCUSSION THREAD

**Discussion by vincerubinetti** [adobe/react-spectrum#5143] [Ideas]
It seems to work when closed, but I'd expect it to work when open or closed, like a native select.

Related discussion on a different library: https://github.com/tailwindlabs/headlessui/discussions/1957

**Comment by LFDanLu** [comment]
Looks like we don't supply a collator to the default keyboard delegate [here](https://github.com/adobe/react-spectrum/blob/db5bb1bdc63abd6a0515591029b4120127fc6dbb/packages/react-aria-components/src/ListBox.tsx#L134-L141), thus making the typeahead results [always null](https://github.com/adobe/react-spectrum/blob/e6d57e7be484639b558b22d99b99b6bb6b4c9b00/packages/%40react-aria/selection/src/useTypeSelect.ts#L79) since we early return [here](https://github.com/adobe/react-spectrum/blob/db5bb1bdc63abd6a0515591029b4120127fc6dbb/packages/%40react-aria/selection/src/ListKeyboardDelegate.ts#L272-L274) if there isn't a collator. We should probably have a default collator setup in RAC ListBox like we have in RSP ListBox, ordinarily the `useListBox`'s default keyboard delegate will have one but we override it with our own keyboard delegate here.

**Reply by vincerubinetti** [reply]
Sounds like this is intended behavior and thus a bug? Maybe should be transferred to an issue?

**Reply by LFDanLu** [reply]
Yep, I've made the issue here: https://github.com/adobe/react-spectrum/issues/5145

**Reply by devongovett** [reply]
Fixed by https://github.com/adobe/react-spectrum/pull/5146
